client/packages/requisitions/src/RequestRequisition/ListView/AppBarButtons.tsx

```diff
@@ -59,15 +60,24 @@ export const AppBarButtons: FC<{
           {t('button.export')}
         </LoadingButton>
       </Grid>
-      <InternalSupplierSearchModal
-        open={modalController.isOn}
+      <CreateRequisitionModal
+        isOpen={modalController.isOn}
         onClose={modalController.toggleOff}
-        onChange={async name => {
+        onChange={async newRequistion => {
           modalController.toggleOff();
-          onCreate({
-            id: FnUtils.generateUUID(),
-            otherPartyId: name?.id,
-          });
+          switch (newRequistion.type) {
+            case 'general':
```

Comment:
:nit I'm surprised not to see an enum for these

Reply:
The telegram conversation swayed me a little bit to use union string type, `.type` would still be restricted to 'general' or 'program', but it's a bit easier to define, and in the downstream PR both types are split between two files, `NewGeneralRequisition` and `NewProgramRequisition`, to use enum we would need to create a common type or have circular dependency. I really don't mind either way, updated in downstream branch (can revert if string union is better):

c22c6dc6887113e7c25ce27acd4350bae22d7aad